examples/12.1.2/docs/react/demo/src/index.tsx

```diff
@@ -46,24 +47,25 @@ const App = () => {
       beforeRenderer={addClassesToRows}
       afterGetRowHeader={drawCheckboxInRowHeaders}
       afterOnCellMouseDown={changeCheckboxCell}
+      beforePaste={omitNonNumericPaste}
```

Comment:
This is to achieve the following requirement:

> I can paste non-numeric values to both the "Progress" and "Rating" columns, and apparently the renderer will try to handle them. Try pasting `12394-3841-34`. When you open the editor of such a cell it is empty.

I think that this implementation is not as good as it could be. The best way to disallow certain inputs is to use a validator function and `allowInvalid: false` configuration option. This is how e.g. date picker does it.